### ISSUE TITLE
Better unit tests before tidying up code

### ISSUE THREAD

**Issue by credfeto** [credfeto/credfeto-enum-source-generation#25]

**Comment by dnyw4l3n13** [comment]
## Status Update (2026-06-19)

### What has been completed

The following test improvements for issue #25 have already been merged into `main`:

| Issue | Assembly | Status |
|---|---|---|
| #98 | `Credfeto.Enumeration.Source.Generation` | ✅ Merged (PR #106) |
| #99 | `Credfeto.Enumeration.Source.Generation.Generics` | ✅ Merged (PR #103) |
| #100 | `Credfeto.Enumeration.Source.Generation.Attributes` | ✅ Merged (PR #102) |
| original | `Credfeto.Enumeration.Source.Generation.Models.Tests` | ✅ Merged (`342a3275`) — added `IsDefinedTests`, improved `GetName`/`GetDescription` coverage |

### What remains

| Issue | Assembly | Status |
|---|---|---|
| #101 | `Credfeto.Enumeration.Source.Generation.Models` | 🔄 Code complete, blocked by infrastructure |

The code for issue #101 is complete and staged on branch `test/101-increase-http-status-code-coverage`:

1. All remaining `HttpStatusCode` values added as `[InlineData]` entries to `GetNameForExternalEnumReturnsMemberName` — covers every switch branch in the generated `GetName(HttpStatusCode)` method.
2. `GetNameForOutOfRangeHttpStatusCodeThrows` — new `[Fact]` asserting `UnreachableException` for `(HttpStatusCode)9999`, covering `ThrowInvalidEnumMemberException`.

### Blocker

The NuGet proxy (`api-nuget.markridgwell.com`) is unreachable in this environment. `dotnet restore` fails with `NU1301`, which means the pre-commit hook (`buildtest`) cannot run. No commit or push can be made until connectivity is restored.

Labelling this issue `Blocked` until the NuGet proxy is reachable again.

**Comment by dnyw4l3n13** [comment]
## Status Update (2026-06-26)

All sub-issues for issue #25 are now complete.

| Issue | Assembly | Status |
|---|---|---|
| #98 | `Credfeto.Enumeration.Source.Generation` | ✅ Merged (PR #106) |
| #99 | `Credfeto.Enumeration.Source.Generation.Generics` | ✅ Merged (PR #103) |
| #100 | `Credfeto.Enumeration.Source.Generation.Attributes` | ✅ Merged (PR #102) |
| original | `Credfeto.Enumeration.Source.Generation.Models.Tests` | ✅ Merged (`342a3275`) |
| #101 | `Credfeto.Enumeration.Source.Generation.Models` | 🔄 PR #112 open, CI running |

The NuGet proxy connectivity issue that previously blocked issue #101 is now resolved. The branch `test/101-increase-http-status-code-coverage` has been pushed and PR #112 opened.

Coverage for `Credfeto.Enumeration.Source.Generation.Models` is now **100% line / 100% branch** (was 27.7% / 48%).

**Comment by dnyw4l3n13** [comment]
## Status Update (2026-07-18)

All work tracked under this issue is now complete:

| Issue | Assembly | Status |
|---|---|---|
| #98 | Credfeto.Enumeration.Source.Generation | Merged (PR #106) |
| #99 | Credfeto.Enumeration.Source.Generation.Generics | Merged (PR #103) |
| #100 | Credfeto.Enumeration.Source.Generation.Attributes | Merged (PR #102) |
| original | Credfeto.Enumeration.Source.Generation.Models.Tests | Merged (342a3275) |
| #101 | Credfeto.Enumeration.Source.Generation.Models | Merged (PR #112, 2026-06-28) |

No further unit-test work remains. I have not written or posted a "## Implementation Plan" comment for #25 itself because there is no remaining implementation to plan - every concrete deliverable was already carried out and merged under the sub-issues above.

Open question for a human: the title is "Better unit tests before tidying up code". On the plain reading, the "tidying up code" work is separate follow-up scope that was never part of this issue's deliverables (no sub-PR references it, and #25's body is empty). Should this issue now be closed as complete, or is there additional "tidying up" scope intended to be done under #25 itself before it closes? If the latter, please describe the desired scope (or link/create a follow-up issue) so a plan can be produced.

Marking this Blocked pending that decision - not because of any CI or infrastructure issue, but because I cannot determine on my own whether #25's remaining scope is "just close it" or "tidy-up work still to be planned". The Workflow board currently shows this item as Development, which no longer reflects reality (no PR or active AI work is in flight for #25); I am leaving it as-is pending your decision so as not to pre-empt the board move that closing/rescoping will require.
